docs/improving-gci-experience.md

```diff
@@ -0,0 +1,377 @@
+working on your problem and try to figure it out. Try not to take more than
+15 minutes of mentor time at a time and ask questions relevant to the issue
+at hand.
+
+* Don't ask for direct answers on code and implementation (e.g. `How should
+this line look exactly?`). Be proactive while solving the problem - mentors
+and other students are here to help you learn, not do the task for you.
+
+## How can I ask mentors to create a task?
+
+If you want to work on something that's not currently a GCI task, e.g. a
+Zulip issue you've found on GitHub, you can ask a mentor to create a task.
+Provide the relevant resources, including the link to the issue.
+
+Because of how the GCI app works, mentors can only create tasks that
+administrators have to publish - they'll ask one of our organization GCI
+admins to publish the task, but it might take a moment before they can get
+around to it.
+
+# How do I go beyond GCI tasks?
+
+GCI is an opportunity for the students to become open-source contributors
+and better programmers. We don't see your contributions as merely fulfilling
+the task requirements, but believe it's a great way to grow and learn the
+skills that will be beneficial for you in the long term.
+
+Whenever you have some idle time, e.g. waiting for a review on your task or
+provisioning your virtual machine, you can spend it productively by learning
+or helping others. This section includes some ideas on what you can do.
+
+## Learning opportunities
+
+* Use Zulip resources:
+  * Read [Zulip documentation](https://zulip.readthedocs.io/en/latest/).
+
+  * Look through [current issues](https://github.com/zulip/zulip/issues),
+  especially the [bite-size
+ones](https://github.com/zulip/zulip/issues?q=is%3Aopen+is%3Aissue+label%3A%22bite+size%22).
+
+  * Read [Tim's post on
+  mypy](http://blog.zulip.org/2016/10/13/static-types-in-python-oh-mypy/).
+
+
+* Use various online courses to help you grow, on topics such as *Python, web
+development, git and GitHub, computer science*. Some examples of free
+courses:
+
+  * [Free Code Camp](https://www.freecodecamp.com/).
+
+  * [Open Source Society University](https://ossu.firebaseapp.com/).
+
+  * [The Hitchhiker's Guide to Python](http://docs.python-guide.org/en/latest/).
+
+* Contribute to other OSS projects - [a list of projects good for
+beginners](https://github.com/MunGell/awesome-for-beginners).
+
+* Find other programming resources that'll help you grow, some examples:
+
+  * Brandon Rhodes's [talk "The Mighty
+  Dictionary"](http://pyvideo.org/pycon-us-2010/the-mighty-dictionary-55.html).
+
+  * Alisson Kaptur's blog posts, examples:
+    * [Effective learning strategies for
+    programmers](http://akaptur.com/blog/2015/10/10/effective-learning-strategies-for-programmers/).
+
+    * [Systematic
+    debugging](http://akaptur.com/blog/2013/07/24/systematic-debugging/).
+
+  * Jessica McKellar's ["Building and Breaking a Python
```

Comment:
You may want to put links to your new blog posts on mentorship in this doc somewhere :)